packages/gatsby/src/schema/resolvers.ts

```diff
@@ -75,34 +54,56 @@ export function findOne<TSource, TArgs>(
 
 type PaginatedArgs<TArgs> = TArgs & { skip?: number; limit?: number }
 
-export function findManyPaginated<TSource, TArgs, TNodeType>(
+export function findManyPaginated<TSource, TArgs>(
   typeName: string
 ): GatsbyResolver<TSource, PaginatedArgs<TArgs>> {
   return async function findManyPaginatedResolver(
-    source,
+    _source,
     args,
     context,
     info
-  ): Promise<IGatsbyConnection<TNodeType>> {
+  ): Promise<IGatsbyConnection<IGatsbyNode>> {
     // Peek into selection set and pass on the `field` arg of `group` and
     // `distinct` which might need to be resolved.
     const group = getProjectedField(info, `group`)
     const distinct = getProjectedField(info, `distinct`)
     const max = getProjectedField(info, `max`)
+    const min = getProjectedField(info, `min`)
+    const sum = getProjectedField(info, `sum`)
+
+    // Apply paddings for pagination
+    // (for previous/next node and also to detect if there is a previous/next page)
+    const skip = typeof args.skip === `number` ? Math.max(0, args.skip - 1) : 0
+    const limit = typeof args.limit === `number` ? args.limit + 1 : undefined
+
     const extendedArgs = {
       ...args,
       group: group || [],
       distinct: distinct || [],
       max: max || [],
+      min: min || [],
+      sum: sum || [],
+      skip,
+      limit,
     }
-
-    const result = await findMany<TSource, PaginatedArgs<TArgs>>(typeName)(
-      source,
-      extendedArgs,
-      context,
-      info
+    if (context.stats) {
+      context.stats.totalRunQuery++
+      context.stats.totalPluralRunQuery++
+    }
```

Comment:
Do you know why we have these stats? Maybe add a comment why it exists? (I know it's from the old code but I like to know :joy:)

Reply:
We add it to telemetry here:

https://github.com/gatsbyjs/gatsby/blob/7c343ffd803c6ada0602678b9a6e6d164a58a4fb/packages/gatsby/src/commands/build.ts#L180-L184

Reply:
Added a comment about stats usage